src/main/scala/mesosphere/marathon/upgrade/StartingBehavior.scala

```diff
@@ -38,29 +39,31 @@ trait StartingBehavior extends ReadinessBehavior { this: Actor with ActorLogging
   final override def receive: Receive = readinessBehavior orElse commonBehavior
 
   def commonBehavior: Receive = {
-    case MesosStatusUpdateEvent(_, taskId, StartErrorState(_), _, `pathId`, _, _, _, `versionString`, _, _) => // scalastyle:off line.size.limit
-      log.warning(s"New task [$taskId] failed during app ${runSpec.id.toString} scaling, queueing another task")
-      instanceTerminated(taskId)
+    case InstanceChanged(id, `version`, `pathId`, Terminal(_), _) =>
```

Comment:
This does not bind to the variable, but matches the value in version and path, so that it only accepts InstanceChanged events for the given pathId and version.